sequence_processing_pipeline/Pipeline.py

```diff
@@ -29,16 +31,26 @@ class Pipeline:
                     None, 'adaptation', 'TRUE', 'UCSD', 'FALSE']
 
     def __init__(self, configuration_file_path, run_id, sample_sheet_path,
-                 output_path, qiita_job_id, config_dict=None):
+                 mapping_file_path, output_path, qiita_job_id,
+                 config_dict=None):
         """
         Initialize Pipeline object w/configuration information.
         :param configuration_file_path: Path to configuration.json file.
         :param run_id: Used w/search_paths to locate input run_directory.
-        :param sample_sheet_path: Path to required sample-sheet.
+        :param sample_sheet_path: Path to sample-sheet.
+        :param mapping_file_path: Path to mapping file.
         :param output_path: Path where all pipeline-generated files live.
         :param qiita_job_id: Qiita Job ID creating this Pipeline.
         :param config_dict: (Optional) Dict used instead of config file.
         """
+        if sample_sheet_path is not None and mapping_file_path is not None:
+            raise PipelineError("sample_sheet_path or mapping_file_path "
+                                "must be defined, but not both.")
+
+        if sample_sheet_path is None and mapping_file_path is None:
+            raise PipelineError("sample_sheet_path or mapping_file_path "
+                                "must be defined, but not both.")
+
```

Comment:
Maybe I missed them but I think there are no tests for this, right?

Reply:
:) That might be true; I encountered that error several times during writing and testing so it probably didn't register that there wasn't a test for it.

Reply:
Ah, actually it's test_additional_constuctor_check(). Let me add one more test for when they're both defined.